browser-interface/packages/shared/world/positionThings.ts

```diff
@@ -185,3 +199,17 @@ function computeComponentValue(x: number | number[]) {
 
   return Math.random() * (max - min) + min
 }
+
+function getSpawnPointWorldPosition(spawnPoint: SpawnPoint)
+{
+  const x = spawnPoint.position.x;
+  const y = spawnPoint.position.y;
+  const z = spawnPoint.position.z;
+
+  if (typeof x === "number" && typeof y === "number" && typeof z === "number")
+  {
+    return new Vector3(x, y ,z); 
+  } else {
+    return new Vector3(x[0], y[0], z[0]); 
```

Comment:
why other indexes aren't being considered?
spawnpoint A : {x: [0,9], y: [0], z: [0,10]}
spawnpoint B : {x: 10, y: 0, z: 0}
if you want to teleport to {x: 6, y:0, z: 0}
spawnpoint B will be set as the spawnpoint but it should be spawnpoint A

Reply:
In your example, you will be teleported at spawn point A since the parcel is 0,0 the position to teleport at is 0.0, parcels are 16x16 and you can't teleport to a specific position.

What we can change is to teleport to the middle of the parcel and use the center of the area spawn point to compare their position.